### DISCUSSION TITLE
Apple M1 Chips: crash with attachment of code signature supplement failed: 3

### DISCUSSION THREAD

**Discussion by hmmmm7** [airsdk/Adobe-Runtime-Support#1013] [General]
Hi All,

My app running on framework 476 is crashing on the new Apple M1, with the error report below.
It seems to occur randomly... 
Does anyone have the same issue?
Any clue?

Thanks in Advance!

Process:               myApp [845]
Path:                  /Applications/myApp.app/Contents/MacOS/myApp
Identifier:            myApp
Version:               ???
Code Type:             X86-64 (Translated)
Parent Process:        ??? [1]
Responsible:           myApp [845]
User ID:               502

Date/Time:             2021-07-27 09:22:11.135 -0400
OS Version:            macOS 11.5 (20G71)
Report Version:        12
Anonymous UUID:        1E8310DA-FD7C-3C04-D897-574CEA6CB01D

Time Awake Since Boot: 680 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       EXC_ARM_BREAKPOINT at 0x00007ffdffea4d38 (brk 1)
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Trace/BPT trap: 5
Termination Reason:    Namespace SIGNAL, Code 0x5
Terminating Process:   exc handler [845]

Application Specific Information:
rosetta error: /var/db/oah/223263137464320_223263137464320/66558040e44317dccc1ec2ff668a670267c88095a7855a3e581ad6d4b07f5845/myApp.aot: attachment of code signature supplement failed: 3

**Comment by ajwfrost** [comment]
Hi

We've had a couple reports of this now; there are some similar issues online that suggest there may just be problems in Apple's Rosetta2 implementation but I'm also wondering whether the app is trying to use JIT, which I believe would cause problems for it.

We'll look at automatically turning off X86_64 JIT if we're running on M1 - there's a manual way to override it already though, we'll need to double-check that this is working and I'll then let you know the workaround..

thanks

**Reply by ajwfrost** [reply]
Hi

Can you please try the following:

1) Create the below folder (you may need admin rights):
`/Library/Application Support/Macromedia`

2) Within this, create a file called `mms.cfg`

3) Add a line to this:
`AS3JIT=0`

thanks

**Comment by hmmmm7** [comment]
Andrew, 

Yes, you sent me the same workaround last week.
1 - On a station, the app crash (impossible to start)
2 - I created the file
3 - try to restart: crash

I didn’t restart the station or anything else.. not sure if it was required.

Thanks, 

David
